pkg/services/guardian/provider.go

```diff
@@ -14,9 +14,21 @@ type Provider struct{}
 func ProvideService(store *sqlstore.SQLStore, ac accesscontrol.AccessControl, permissionsServices accesscontrol.PermissionsServices, features featuremgmt.FeatureToggles) *Provider {
 	if features.IsEnabled(featuremgmt.FlagAccesscontrol) {
 		// TODO: Fix this hack, see https://github.com/grafana/grafana-enterprise/issues/2935
-		New = func(ctx context.Context, dashId int64, orgId int64, user *models.SignedInUser) DashboardGuardian {
-			return NewAccessControlDashboardGuardian(ctx, dashId, user, store, ac, permissionsServices)
-		}
+		InitACLGuardianWithStore(store, ac, permissionsServices)
+	} else {
+		InitGuardianWithStore(store)
```

Comment:
This naming isn't ideal, both guardian implementations uses some form of ACL, and we've never referred to fine-grained access control as an ACL. It's mostly called "AccessControl" spelt out, "AC", or "FGAC" (though not so much in code and not in documentation).

`InitLegacyGuardian` and `InitAccessControlGuardian` maybe? (I'm proposing not having "with store" in the name, as that's already in the signature and there isn't an implementation not using the store)